### PR TITLE
Add cluster info with status property

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/3260

### CODE REVIEW

File: service/collector/cluster_info.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	)
+)
+
+func (c *Collector) collectClusterInfo(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics for clusters")
+
+	opts := v1.ListOptions{}
+	clusters, _ := c.g8sClient.ProviderV1alpha1().AWSConfigs("").List(opts)
+
+	for _, cluster := range clusters.Items {
```
Comment (tuommaki) [inline service/collector/cluster_info.go]: Should there be paging for listing? Here is an example I've implemented: https://github.com/giantswarm/cluster-operator/blob/master/pkg/v3/resource/certconfig/current.go#L37-L63

File: service/service.go [hunk 2]
```diff
@@ -211,7 +211,8 @@ func New(config Config) (*Service, error) {
 	var metricsCollector *collector.Collector
 	{
 		c := collector.Config{
-			Logger: config.Logger,
+			Logger:    config.Logger,
+			G8sClient: g8sClient,
```
Comment (tuommaki) [inline service/service.go]: Maybe sort alphabetically?

File: service/collector/collector.go [hunk 3]
```diff
@@ -34,11 +38,15 @@ func New(config Config) (*Collector, error) {
...
+	if config.G8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
 
 	awsClients := awsutil.NewClients(config.AwsConfig)
 
 	c := &Collector{
-		logger: config.Logger,
+		logger:    config.Logger,
+		g8sClient: config.G8sClient,
```
Comment (tuommaki) [inline service/collector/collector.go]: Maybe sort alphabetically?

File: service/collector/collector.go [hunk 4]
```diff
@@ -6,17 +6,21 @@ import (
...
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
 )
 
 type Config struct {
-	Logger micrologger.Logger
+	Logger    micrologger.Logger
+	G8sClient versioned.Interface
```
Comment (tuommaki) [inline service/collector/collector.go]: Maybe sort alphabetically?

File: service/collector/collector.go [hunk 5]
```diff
@@ -6,17 +6,21 @@ import (
...
 
 	AwsConfig        awsutil.Config
 	InstallationName string
 }
 
 type Collector struct {
-	logger micrologger.Logger
+	logger    micrologger.Logger
```
Comment (tuommaki) [inline service/collector/collector.go]: Maybe sort alphabetically?

File: service/collector/collector.go [hunk 6]
```diff
@@ -34,11 +38,15 @@ func New(config Config) (*Collector, error) {
 	if config.InstallationName == "" {
 		return nil, microerror.Maskf(invalidConfigError, "%T.InstallationName must not be empty", config)
 	}
+	if config.G8sClient == nil {
```
Comment (tuommaki) [inline service/collector/collector.go]: Maybe sort alphabetically (i.e. lift this check one up)?

File: service/collector/cluster_info.go (hunk 1 above)
Comment (pipo02mix) [reply service/collector/cluster_info.go]: yeah you are right. I will do

File: service/collector/cluster_info.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+	clustersDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(Namespace, "", "cluster_info"),
+		"Cluster information.",
+		[]string{
+			clusterLabel,
+			statusLabel,
+		},
+		nil,
+	)
+)
```
Comment (kopiczko) [inline service/collector/cluster_info.go]: Nitpick: why not just

```go
var  clustersDesc *prometheus.Desc = prometheus.NewDesc(
	prometheus.BuildFQName(Namespace, "", "cluster_info"),
	"Cluster information.",
	[]string{
		clusterLabel,
		statusLabel,
	},
	nil,
)
```

I personally don't see a reason to align such initializations.

**Pull request merged**